projects/aws/upgrader/upgrade/upgrader.go

```diff
@@ -0,0 +1,118 @@
+package upgrade
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/aws/eks-anywhere-build-tooling/tools/version-tracker/pkg/util/logger"
+)
+
+type Upgrader struct {
```

Comment:
Naming is hard and I recommend the following:

Structs as plain nouns: API, Replica, Object
Interfaces are active nouns: Reader, Writer, JobProcessor
Functions and methods are verbs: Read, Process, Sync 

ref: https://peter.bourgon.org/blog/2019/04/24/go-naming-tips.html